### DISCUSSION TITLE
how do i hide players

### DISCUSSION THREAD

**Discussion by malafk** [Minestom/Minestom#1332] [Help] [answered]
help

**Comment by ryleu** [comment]
```java
player.setInvisible(true);
```

**Reply by malafk** [reply]
does this hide from tablist aswell, as I want to reduce bandwidth and player lag etc

**Reply by GoldenCookieGIT** [reply]
to hide from tablist you send a player info packet with the remove action

**Comment by KrystilizeNevaDies** [comment, marked as answer]
To hide players, you can use a Viewer predicate, something like 
```java
player.updateViewableRule(otherPlayer -> false);
```

**Reply by malafk** [reply]
what would be the best way to do it for all players

**Reply by KrystilizeNevaDies** [reply]
On PlayerSpawnEvent, set the predicate for the player

**Reply by malafk** [reply]
I am not really sure how predicates work can you show an example?

**Reply by KrystilizeNevaDies** [reply]
A predicate is just a function that takes something in and returns true of false.
Example:
```java
// Only show this player to the other player if they have the same exp level
player.updateViewableRule(otherPlayer -> {
    return player.getExperienceLevel() == otherPlayer.getExperienceLevel();
});
```
